### PR TITLE
Version 17.0

### PR DESCRIPTION
**Pull request by def-**
DDNet 17.0 is supposed to release in 1 week, assuming no bad bugs are found. Please test the Release Candidate to prevent problems being only discovered after release. Report bugs in the #bugs channel on DDNet Discord or directly on Github:

- Windows 32bit: https://ddnet.tw/downloads/DDNet-17.0-rc3-win32.zip
- Windows 64bit: https://ddnet.tw/downloads/DDNet-17.0-rc3-win64.zip
- Linux x86: https://ddnet.tw/downloads/DDNet-17.0-rc3-linux_x86.tar.xz
- Linux x86-64: https://ddnet.tw/downloads/DDNet-17.0-rc3-linux_x86_64.tar.xz
- macOS: https://ddnet.tw/downloads/DDNet-17.0-rc3-macos.dmg

You can also test the release candidate in Steam by clicking on Settings -> Properties -> Betas and opting into the Release Candidates. This will automatically update you to a new release candidate when it is available, and to the next release when no new candidate is available. So you can always leave this setting enabled to help us test new releases earlier.

The following changes are included, ideally test their behaviour specifically. But just playing as you do normally also helps:
killfeed
friendlist

https://github.com/ddnet/ddnet/assets/2335377/8ec7183c-8eee-445b-b6fc-2f1a4dfef567

proof

- [Client] **Condensed team killfeed** [Voxel]
- [Client] **Improved friends list UI** [Robyt3]
- [Client] **Improved line input and IME support** [Robyt3]
- [Editor] **Add proof mode for menu backgrounds** [Marmare314]
- [Client] Fix cl_show_others_alpha not used in places where it can be [Ryozuki]
- [Client] Show error message popup on assertion error in client and when client fails to launch [Robyt3]
- [Client] Multiple broadcast rendering fixes [Robyt3 & Jupeyy]
- [Client] Add the DDNet git hash and "is steam" to the crash file [Jupeyy]
- [Client] Fix client crash on team kill message with invalid team [Robyt3]
- [Client] Fix client crash when spectating on server with missing game info [Robyt3]
- [Client] Fix several issues with the text rendering [Robyt3]
- [Client] Highlight 0XF game type [0xfaulty]
- [Client] Fix DDNet freezing on resize on macOS [Ryozuki]
- [Client] Quit client when existing config file cannot be loaded, add restart command to client [Robyt3]
- [Client] Fix client crash/hang when launching without data directory [Robyt3]
- [Client] Add missing attribute shield entities for front and switch layers [Robyt3]
- [Editor] Improve layout of new folder, map details and event popups [Robyt3]
- [Editor] Improve editor file browser button layout when screen width is low [Robyt3]
- [Editor] Fix transparency in grass_main [mind]
- [Editor] Don't render switch number and delay for tiles where they are unused [Robyt3]
- [Editor] Make envelope editor resizeable by dragging [Marmare314]
- [Editor] Move some buttons into menus [Marmare314]
- [Server] Fix time score [Ryozuki]
- [Server] Fix swapping with dragger beams and plasmas [Robyt3]
- [Server] Fix hook prediction [trml]
- [Server] Allow using rescue (/r) on health pickup [Robyt3]
- [Server] Add client_score_kind field to serverinfo [Ryozuki]

### CODE REVIEW

**Comment by def-** [comment]
Oh no, win32 fails, but win64 works:
```
[ 73%] Linking CXX executable DDNet.exe
/usr/lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld: CMakeFiles/game-client.dir/objects.a(http.cpp.obj): in function `ZN12CHttpRequestD2Ev':
/home/deen/isos/ddnet/ddnet-source/src/engine/shared/http.cpp:140: undefined reference to `_imp__curl_slist_free_all'
/usr/lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld: CMakeFiles/game-client.dir/objects.a(http.cpp.obj): in function `Z8HttpInitP8IStorage':
/home/deen/isos/ddnet/ddnet-source/src/engine/shared/http.cpp:76: undefined reference to `_imp__curl_global_init'
/usr/lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld: /home/deen/isos/ddnet/ddnet-source/src/engine/shared/http.cpp:80: undefined reference to `_imp__curl_share_init'
/usr/lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld: /home/deen/isos/ddnet/ddnet-source/src/engine/shared/http.cpp:87: undefined reference to `_imp__curl_version_info'
/usr/lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld: /home/deen/isos/ddnet/ddnet-source/src/engine/shared/http.cpp:95: undefined reference to `_imp__curl_share_setopt'
/usr/lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld: CMakeFiles/game-client.dir/objects.a(http.cpp.obj): in function `Z9EscapeUrlPciPKc':
/home/deen/isos/ddnet/ddnet-source/src/engine/shared/http.cpp:114: undefined reference to `_imp__curl_easy_escape'
/usr/lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld: /home/deen/isos/ddnet/ddnet-source/src/engine/shared/http.cpp:116: undefined reference to `_imp__curl_free'
/usr/lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld: CMakeFiles/game-client.dir/objects.a(http.cpp.obj): in function `Z19HttpHasIpresolveBugv':
/home/deen/isos/ddnet/ddnet-source/src/engine/shared/http.cpp:123: undefined reference to `_imp__curl_version_info'
/usr/lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld: CMakeFiles/game-client.dir/objects.a(http.cpp.obj): in function `ZN12CHttpRequest7RunImplEPv':
/home/deen/isos/ddnet/ddnet-source/src/engine/shared/http.cpp:196: undefined reference to `_imp__curl_easy_setopt'
/usr/lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld: /home/deen/isos/ddnet/ddnet-source/src/engine/shared/http.cpp:257: undefined reference to `_imp__curl_version_info'
/usr/lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld: CMakeFiles/game-client.dir/objects.a(http.cpp.obj): in function `ZN12CHttpRequest6HeaderEPKc':
/home/deen/isos/ddnet/ddnet-source/src/engine/shared/http.cpp:397: undefined reference to `_imp__curl_slist_append'
/usr/lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld: CMakeFiles/game-client.dir/objects.a(http.cpp.obj): in function `ZN12CHttpRequest7RunImplEPv':
/home/deen/isos/ddnet/ddnet-source/src/engine/shared/http.cpp:295: undefined reference to `_imp__curl_easy_perform'
/usr/lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld: CMakeFiles/game-client.dir/objects.a(http.cpp.obj): in function `ZN12CHttpRequest6HeaderEPKc':
/home/deen/isos/ddnet/ddnet-source/src/engine/shared/http.cpp:397: undefined reference to `_imp__curl_slist_append'
/usr/lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld: CMakeFiles/game-client.dir/objects.a(http.cpp.obj): in function `ZN12CHttpRequest3RunEv':
/home/deen/isos/ddnet/ddnet-source/src/engine/shared/http.cpp:160: undefined reference to `_imp__curl_easy_init'
/usr/lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld: /home/deen/isos/ddnet/ddnet-source/src/engine/shared/http.cpp:162: undefined reference to `_imp__curl_easy_cleanup'
/usr/lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld: CMakeFiles/game-client.dir/objects.a(http.cpp.obj): in function `ZN12CHttpRequest6HeaderEPKc':
/home/deen/isos/ddnet/ddnet-source/src/engine/shared/http.cpp:397: undefined reference to `_imp__curl_slist_append'
collect2: error: ld returned 1 exit status
```

**Comment by def-** [comment]
Fixed

**Comment by Jupeyy** [comment]
wanted to merge libs, ups

File: data/languages/german.txt [hunk 1]
```diff
@@ -1708,3 +1708,28 @@ Open the settings file
 
 Create a random skin
 == Erstelle einen zufälligen Skin
+
+Online players (%d)
+== Online-Spieler (%d)
+
+Online clanmates (%d)
+== Online-Clanmates (%d)
```
Comment (murpii) [inline data/languages/german.txt:1716]: Clanmitglieder ? ^^

**Comment by Robyt3** [comment]
You also need to update `src/game/version.h`

**Comment by def-** [comment]
Update in what way? https://github.com/ddnet/ddnet/pull/6620/files#diff-c2aba830bcc905e7db2c01a9fdaa5228d6b8169bb47f0991faafff96bc947f12 is not enough?

**Comment by Robyt3** [comment]
Ah, I looked on master and missed the commit here. We need to remember to also add that commit on master.

**Comment by def-** [comment]
Why? Is there a feature that is version dependent? If so, I can do it now.

**Comment by Robyt3** [comment]
> Why? Is there a feature that is version dependent? If so, I can do it now.

I don't think so. Team kill messages are not version dependent I think. We forgot updating the version a while for the last update though, so I just want to make sure master is also on the latest version.

**Comment by def-** [comment]
I like to do it on the day of the release, but sometimes forget. Maybe we could add a CI check that verifies it, but that seems annoying to maintain

**Pull request closed without merging**